lang/typescript/src/extended-address/generateAddress1.test.ts

```diff
@@ -0,0 +1,121 @@
+import {generateAddress1} from './generateAddress1';
+
+describe('generateAddress1', () => {
+  test('returns null if address1 is missing for standard region', () => {
+    expect(
+      generateAddress1({
+        countryCode: 'US',
+        streetName: 'Main',
+        streetNumber: '123',
+      }),
+    ).toBeNull();
+  });
+
+  test('returns null if address1, streetName, and streetNumber are missing for region with additional fields', () => {
+    expect(generateAddress1({countryCode: 'BR'})).toBeNull();
+    expect(generateAddress1({countryCode: 'CL'})).toBeNull();
+  });
+
+  test('returns address1 if defined', () => {
+    expect(
+      generateAddress1({
+        countryCode: 'US',
+        address1: '123 Main',
+        streetName: 'Main',
+        streetNumber: '123',
+      }),
+    ).toBe('123 Main');
+    expect(
+      generateAddress1({
+        countryCode: 'BR',
+        address1: 'Main 123',
+        streetName: 'Main',
+        streetNumber: '123',
+      }),
+    ).toBe('Main 123');
+    expect(
+      generateAddress1({
+        countryCode: 'CL',
+        address1: 'Main 123',
+        streetName: 'Main',
+        streetNumber: '123',
+      }),
+    ).toBe('Main 123');
```

Comment:
I think this is wrong. If a country supports the extended format the additional fields should have higher priority than the standard fields.

With this logic, I have to make sure that every time I use this function I have to remove the previous `address1`. Otherwise the function won't do anything and I'll stay with the old `address1`.

Reply:
> This is expected, yes.

Yes, I understand it was implemented on purpose. But I think it's wrong. 

I think passing a full address so the library takes care of using the correct fields to generate an address1 is ok. But the library is meant to be used with the additional fields and the result that the function generates in this case is wrong. The correct address1 in that case should include the reserved delimiter.

> If there's an existing address1 you're migrating an existing address to the new format it should be split out first.

It doesn't necessarily have to be that we are migrating. Let's say I have the following values (using additional fields):

```js
{
  address1: "Main\u00A0123";
  streetName: "Main";
  streetNumber: "123";
  ...rest
}

```

Now the user writes a new value to the streetNumber field, for example: `456`

I call then this function to generate the new address1 (which I cannot update without calling this function first):

```js
generateAddress1({
  address1: "Main\u00A0123",
  streetName: "Main",
  streetNumber: "456",
  ...rest
});
```

I would get a wrong result. So for me the library wouldn't work as expected. 

If we ship it like this we need to make it sure that developers always clear the `address1` value of the address before calling this function.

Reply:
We need to always store `address1`. Outside of checkout `streetName` and `streetNumber` do not exist. An even inside checkout there are places where they do not exist like wallets or extensions.

Reply:
Followup after a discussion with @rochlefebvre and @josemiguel-alvarez:

I will modify to follow this set of rules:
- Region does not define `combined_address_field`: return `address1` unmodified (no change)
- Region does define `combined_address_field`, but each extended field is `undefined`: return `address1` unmodified (no change)
- Region does define `combined_address_field`, but any extended field is a string (empty or otherwise): return concatenated address1 (⚠️ change)
- No relevant fields defined: Return `null` (no change)